src/core/lombok/javac/handlers/HandleWithBy.java

```diff
@@ -0,0 +1,336 @@
+		boolean makeAbstract = typeNode != null && typeNode.getKind() == Kind.TYPE && (((JCClassDecl) typeNode.get()).mods.flags & Flags.ABSTRACT) != 0;
+		
+		if (fieldNode.getKind() != Kind.FIELD) {
+			fieldNode.addError("@WithBy is only supported on a class or a field.");
+			return;
+		}
+		
+		JCVariableDecl fieldDecl = (JCVariableDecl) fieldNode.get();
+		String methodName = toWithByName(fieldNode);
+		
+		if (methodName == null) {
+			fieldNode.addWarning("Not generating a withXBy method for this field: It does not fit your @Accessors prefix list.");
+			return;
+		}
+		
+		if ((fieldDecl.mods.flags & Flags.STATIC) != 0) {
+			if (strictMode) fieldNode.addWarning("Not generating " + methodName + " for this field: WithBy methods cannot be generated for static fields.");
+			return;
+		}
+		
+		if ((fieldDecl.mods.flags & Flags.FINAL) != 0 && fieldDecl.init != null) {
+			if (strictMode) fieldNode.addWarning("Not generating " + methodName + " for this field: WithBy methods cannot be generated for final, initialized fields.");
+			return;
+		}
+		
+		if (fieldDecl.name.toString().startsWith("$")) {
+			if (strictMode) fieldNode.addWarning("Not generating " + methodName + " for this field: WithBy methods cannot be generated for fields starting with $.");
+			return;
+		}
+		
+		for (String altName : toAllWithByNames(fieldNode)) {
+			switch (methodExists(altName, fieldNode, false, 1)) {
+			case EXISTS_BY_LOMBOK:
+				return;
+			case EXISTS_BY_USER:
+				if (strictMode) {
+					String altNameExpl = "";
+					if (!altName.equals(methodName)) altNameExpl = String.format(" (%s)", altName);
+					fieldNode.addWarning(
+						String.format("Not generating %s(): A method with that name already exists%s", methodName, altNameExpl));
+				}
+				return;
+			default:
+			case NOT_EXISTS:
+				//continue scanning the other alt names.
+			}
+		}
+		
+		long access = toJavacModifier(level);
+		
+		JCMethodDecl createdWithBy = createWithBy(access, fieldNode, fieldNode.getTreeMaker(), source, onMethod, makeAbstract);
+		recursiveSetGeneratedBy(createdWithBy, source);
+		injectMethod(typeNode, createdWithBy);
+	}
+	
+	private static final LombokImmutableList<String> NAME_JUF_FUNCTION = LombokImmutableList.of("java", "util", "function", "Function");
+	private static final LombokImmutableList<String> NAME_JUF_OP = LombokImmutableList.of("java", "util", "function", "UnaryOperator");
+	private static final LombokImmutableList<String> NAME_JUF_DOUBLEOP = LombokImmutableList.of("java", "util", "function", "DoubleUnaryOperator");
+	private static final LombokImmutableList<String> NAME_JUF_INTOP = LombokImmutableList.of("java", "util", "function", "IntUnaryOperator");
+	private static final LombokImmutableList<String> NAME_JUF_LONGOP = LombokImmutableList.of("java", "util", "function", "LongUnaryOperator");
+	
+	public JCMethodDecl createWithBy(long access, JavacNode field, JavacTreeMaker maker, JavacNode source, List<JCAnnotation> onMethod, boolean makeAbstract) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 32. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.